invalidPropsals.ts

```diff
@@ -0,0 +1,145 @@
+                        invalidProposalsInSpace.push(proposal.id)
+                        if(result.reason === 'not_by_member') {
+                            console.log(proposal.id + " - proposal not created by a member - marked it as Invalid");
+                        } else if(result.reason === 'min_score') {
+                            console.log(proposal.id + " - less than min score - marked it as Invalid");
+                        }
+                    } else {
+                        console.log(proposal.id + " - is valid proposal")
+                    }
+                } catch(error) {
+                    console.log(error)
+                }
+            }));
+            await markProposalsInvalid(invalidProposalsInSpace);
+        }
+    } catch(error) {
+        console.log(error)
+    }
+}
+
+const verifyProposal = (proposal) => {
+    return new Promise(async (resolve, reject) => {
+        const space = spaces[proposal.space];
+        const result = {invalid: false, reason: ""}
+        if(!space) {
+            return reject(proposal.id + " - No Space found - " + proposal.space + " - Ignoring");
+        }
+        const members = space.members
+          ? space.members.map(address => address.toLowerCase())
+          : [];
+        const isMember = members.includes(proposal.address.toLowerCase());
+        
+        if (space.filters && space.filters.onlyMembers && !isMember) {
+            result.invalid = true;
+            result.reason = "not_by_member";
+            return resolve(result);
+        } else if (!isMember && space.filters && space.filters.minScore) {
+            try {
+                const scores = await snapshot.utils.getScores(
+                    proposal.space,
+                    space.strategies,
+                    space.network,
+                    snapshot.utils.getProvider(space.network),
+                    [proposal.address]
+                );
+                const totalScore: any = scores
+                    .map((score: any) =>
+                    Object.values(score).reduce((a, b: any) => a + b, 0)
+                    )
+                    .reduce((a, b: any) => a + b, 0);
+
+                if (totalScore < space.filters.minScore) {
+                    result.invalid = true;
+                    result.reason = "min_score";
+                    return resolve(result);
+                };
+            } catch (e) {
+                return reject(proposal.id + " - failed to check voting power - ignored");
+            }
+        }
+
+        return resolve(result);
+    });
+}
+
+const loadSpaces = async () => {
+    try {
+        const data = await fetch(spacesURL);
+        return data.json();
+    } catch(error) {
+        console.log(error);
+    }
+}
+
+const markProposalsInvalid = async (invalidProposalsInSpace) => {
+    try {
+        if(invalidProposalsInSpace.length > 0) {
+            const query = `UPDATE ${tableName} SET type = ? WHERE id IN (?)`;

```

Comment:
Can you add WHERE type = 'proposal' to make sure we don't change anything else. Also you can use directly `messages` instead of `${tableName}`, we don't change table in this script we don't need to have it dynamic, it will make the script more readable.